js/ui/appDisplay.js

```diff
@@ -2406,7 +2406,7 @@ var AppIcon = GObject.registerClass({
         let view = _getViewFromIcon(source);
 
         return source != this &&
-               (source instanceof AppIcon) &&
```

Comment:
Note that the class expression is named `AppDisplayIcon` and not `AppIcon` here. Class declarations and class expressions do have a special case for accessing the class name inside the class body. Sadly this is broken for GObject classes because of the `GObject.registerClass()` call wrapping the class expression.

You could work around it by moving the definition of `_canAccept` outside of the class body, as is done [here](https://gitlab.gnome.org/GNOME/gjs/blob/master/installed-tests/js/testGtk.js#L51-76):
```js
AppIcon.prototype._canAccept = function (source) {
    ...
    if (source instanceof AppIcon) ...
};
```
I bring this up because in theory the `this.constructor` solution will break if you have two different subclasses of AppIcon ("A" and "B") and you drag an A icon onto a B icon. However, I don't think AppIcon is actually subclassed in the current codebase, so you might just be able to ignore this.

Reply:
Turns out we have a subclass of `AppIcon`: the `HackIcon`.

Reply:
I simply removed this commit; things won't necessarily break without it.

Reply:
Oh, interesting, I grepped for `extends AppIcon` but maybe I didn't check thoroughly enough or maybe I checked on an upstream branch... Yes, that would break if dragging the Hack icon onto the app center icon :smile: